src/Toml/Parser.hs

```diff
@@ -159,10 +161,19 @@ tableNameP = lexeme $ between (char '[') (char ']') keyP
 
 -- Values
 
+decimalP :: Parser Integer
+decimalP = mkInteger <$> decimalStringP
+  where
+    decimalStringP   = fst <$> match (some digitChar >> many _digitsP)
+    _digitsP         = try (char '_' >> some digitChar)
```

Comment:
Shouldn't it be okay to wrap only `char '_'` in `try`? My personal feeling is that `try` can decrease performance a lot, so it's better to reduce scope of `try` parser combinator as much as possible.

Reply:
I tried with it just on the `char '_'` but then I get a parse error on strings that end with an `_` :(

Reply:
@tomphp From the TOML specification:

> For large numbers, you may use underscores between digits to enhance readability

So it's okay if parser can't parsee number like `13_`. Underscores are allowed only between characters.

Reply:
Done